src/builders/CommandStep.ts

```diff
@@ -65,19 +72,70 @@ export class CommandStep implements StepBuilder, KeyBuilder, LabelBuilder, Condi
     this.#plugins.push(plugin)
     return this
   }
+  
+  parallelism(parallelism: number): this {
+    this.#parallelism = parallelism
+    return this
+  }
+  
+  env(env: Record<string, string | number>): this {
+    this.#env = env
+    return this
+  }
 
-  build(): CommandStepSchema {
-    if (!this.#command) {
-      throw new Error('CommandStep requires a command.')
+  addEnv(key: string, value: string | number): this {
+    if (!this.#env) {
+      this.#env = {}
+    }
+    this.#env[key] = value
+    return this
+  }
+
+  concurrency(jobs?: number, group?: string): this {
+    this.#concurrency = jobs
+    this.#concurrency_group = group
+    return this
+  }
+
+  softFail(fail: boolean = true): this {
+    this.#soft_fail = fail
+    return this
+  }
+
+  timeout(minutes: number): this {
+    this.#timeout_in_minutes = minutes
+    return this
+  }
+
+  artifactPath(path: string): this {
+    if (!this.#artifact_paths) {
+      this.#artifact_paths = []
     }
+    this.#artifact_paths.push(path)
+    return this
+  }
+
+  build(): CommandStepSchema {
+    const commandKey = this.#command && Array.isArray(this.#command) && this.#command.length > 1 ? 'commands' : 'command'
     const object: CommandStepSchema = {
-      command: this.#command,
+      ...{ [commandKey]: this.#command },
       ...this.#keyHelper.build(),
       ...this.#labelHelper.build(),
       ...this.#conditionHelper.build(),
       ...this.#branchesHelper.build(),
       ...this.#dependenciesHelper.build(),
       ...this.#skipHelper.build(),
+      ...(this.#parallelism ? { parallelism: this.#parallelism } : {}),
+      ...(this.#concurrency ? {concurrency: this.#concurrency} : {}),
+      ...(this.#concurrency_group ? {concurrency_group: this.#concurrency_group} : {}),
+      ...(this.#env ? {env: this.#env} : {}),
+      ...(this.#soft_fail ? {soft_fail: this.#soft_fail} : {}),
+      ...(this.#timeout_in_minutes ? {timeout_in_minutes: this.#timeout_in_minutes} : {}),
+      ...(this.#artifact_paths ? {artifact_paths: this.#artifact_paths} : {}),
+    }
+
+    if (!this.#command || (Array.isArray(this.#command) && this.#command.length === 0)) {
+      throw new Error('CommandStep requires a command.')
```

Comment:
Definitely what the docs suggest but does it actually? Isn't there times we've just used the docker plugin command?

Reply:
Removing this for now